src/service/fixed-layer.js

```diff
@@ -306,37 +321,23 @@ export class FixedLayer {
   }
 
   /**
-   * Apply or reset transform style to fixed elements. The existing transition,
-   * if any, is disabled when custom transform is supplied.
-   * @param {?string} transform
+   * Apply transform style to all fixed elements.
+   * @param {string=} opt_transform undefined to reset.
    */
-  transformMutate(transform) {
-    // Unfortunately, we can't do anything with sticky elements here. Updating
-    // `top` in animation frames causes reflow on all platforms and we can't
-    // determine whether an element is currently docked to apply transform.
-    if (transform) {
-      // Apply transform style to all fixed elements
-      this.elements_.forEach(e => {
-        if (e.fixedNow && e.top) {
-          setStyle(e.element, 'transition', 'none');
-          if (e.transform && e.transform != 'none') {
-            setStyle(e.element, 'transform', e.transform + ' ' + transform);
-          } else {
-            setStyle(e.element, 'transform', transform);
-          }
-        }
-      });
-    } else {
-      // Reset transform style to all fixed elements
-      this.elements_.forEach(e => {
-        if (e.fixedNow && e.top) {
-          setStyles(e.element, {
-            transform: '',
-            transition: '',
-          });
-        }
-      });
-    }
+  transformMutate(opt_transform) {
```

Comment:
I still think this should be moved out of fixed-layer and into viewport-impl. All that's needed is to return an array of currently fixed-top elements, and iterate of that list. FixedLayer itself has no concern with `transform` styles, there's really no reason for it to live here.